glib-macros/src/enum_derive.rs

```diff
@@ -29,21 +27,14 @@ fn gen_enum_values(
         let mut value_name = name.to_string().to_upper_camel_case();
         let mut value_nick = name.to_string().to_kebab_case();
 
-        let attrs = parse_item_attributes("enum_value", &v.attrs);
-        let attrs = match attrs {
-            Ok(attrs) => attrs,
-            Err(e) => abort_call_site!(
-                "{}: derive(glib::Enum) enum supports only the following optional attributes: #[enum_value(name = \"The Cat\", nick = \"chat\")]",
-                e
-            ),
-        };
-
-        attrs.into_iter().for_each(|attr|
-            match attr {
-                ItemAttribute::Name(n) => value_name = n,
-                ItemAttribute::Nick(n) => value_nick = n,
-            }
-        );
+        let mut name_attr = NestedMetaItem::<syn::LitStr>::new("name").value_required();
+        let mut nick = NestedMetaItem::<syn::LitStr>::new("nick").value_required();
+
+        // the attribute `enum_value` is optional, we don't need to check if we found it
+        let _ = parse_nested_meta_items(&v.attrs, "enum_value", &mut [&mut name_attr, &mut nick]);
```

Comment:
Even if the attribute is optional, I have to handle the case when the attribute has been found but the parsing of it has failed.

I need a match statement here.